tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -179,24 +228,9 @@ static void gatt_scan_cb(const bt_addr_le_t *addr, int8_t rssi,
 		parse_mesh_proxy_service(buf);
 	}
 
-	if (cnt) {
-		int64_t delta = k_uptime_delta(&timestamp);
-
-		LOG_INF("rx: %s +%d ms", txt_msg, delta);
-
-		ASSERT_TRUE(delta >= gatt_param.interval &&
-			delta < (gatt_param.interval + 15));
-	} else {
-		timestamp = k_uptime_get();
+	LOG_INF("rx: %s", txt_msg);
 
-		LOG_INF("rx: %s +0 ms", txt_msg);
-	}
-
-	cnt++;
-
-	if (cnt == gatt_param.transmits) {
-		cnt = 0;
-		timestamp = 0;
+	if (check_delta_time(gatt_param.transmits-1, gatt_param.interval, 15)) {
```

Comment:
Why it can't be:
```suggestion
	if (check_delta_time(gatt_param.transmits, gatt_param.interval, 15)) {
```
with `if (cnt == gatt_param.transmits) {` in `check_delta_time`?

Reply:
I tried to avoid using any of the gatt_param or xmit_param inside the check_delta_time function. It uses only input values.
Then I have to do a +1 for "if (check_delta_time(xmit_param.retr+1, xmit_interval)) {" and I fell that it is safer code that the function has "if (cnt > retr) {" in the function than "if (cnt == transmits) {". Could use "if (cnt >= transmits), but then we still have to use +1 on the places xmit_param.retr is used. I do not see that as better.

Reply:
What if increment `transmits` by one in `test_rx_proxy_mixin`?

Reply:
decrement maybe? As I wrote in another comment then it is already a retransmit but not transmit.
https://github.com/zephyrproject-rtos/zephyr/pull/50296#discussion_r986614633